src/plus/gk/account/authenticationConnection.ts

```diff
@@ -69,25 +69,24 @@ export class AuthenticationConnection implements Disposable {
 		this._pendingStates.set(scopeKey, [...existingStates, gkstate]);
 
 		const callbackUri = await env.asExternalUri(
-			Uri.parse(
-				`${env.uriScheme}://${this.container.context.extension.id}/${AuthenticationUriPathPrefix}?gkstate=${gkstate}`,
-			),
+			Uri.parse(`${env.uriScheme}://${this.container.context.extension.id}/${AuthenticationUriPathPrefix}`),
 		);
 
-		const uri = this.connection.getAccountsUri(
-			'register',
-			`${scopes.includes('gitlens') ? 'referrer=gitlens&' : ''}pass-token=true&return-url=${encodeURIComponent(
-				callbackUri.toString(),
-			)}`,
+		const uri = this.connection.getGkDevAccountsUri(
+			signUp ? 'register' : 'login',
+			`${scopes.includes('gitlens') ? 'source=gitlens&' : ''}state=${encodeURIComponent(
+				gkstate,
+			)}&redirect_uri=${encodeURIComponent(callbackUri.toString())}`,
 		);
-		void (await env.openExternal(uri));
+		// @ts-expect-error openExternal does allow a string, though the typing doesn't reflect it. Must pass as a string to avoid a double encoding
```

Comment:
Rather than using // @ts-expect-error, we could cast, as we do here:

https://github.com/gitkraken/vscode-gitlens/blob/690e17403f098e00bb3d3dc61e645ce8128598e3/src/git/remotes/remoteProvider.ts#L172-L174

Reply:
We could also add that to a helper function in system/utils.ts or something too

Reply:
Done in commit: [Switches to existing fn for opening external URL](https://github.com/gitkraken/vscode-gitlens/pull/3089/commits/43f057c4993c87c5c842d8d8a7a4b476e417bd4f)